commit 23d6d850c0e81644ccaf5a4ecc5370c81c3f79fc (refs/changes/35/335/3)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-08-24T00:32:52+00:00 (3 years, 2 months ago)
    
    MANTA-2966 binder should check names for validity before passing to ZK client
    Reviewed by: Dave Pacheco <dap@joyent.com>

diff --git a/lib/server.js b/lib/server.js
index d9cc2d7..6821828 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -130,6 +130,16 @@ function resolve(options, query, cb) {
                 return;
         }
 
+        domain = domain.toLowerCase();
+        if (/[^a-z0-9_.-]/.test(domain)) {
+                req.log.debug('request for an invalid name: this client is ' +
+                    'probably misbehaving');
+                query.setError('eserver');
+                query.respond();
+                cb();
+                return;
+        }
+
         req.domain = domain;
         zk.resolveName(req, function (err, record) {
                 if (err && (typeof (err.getCode) !== 'function' ||
